lib/const.rb

```diff
@@ -305,6 +305,7 @@
 CREW_PERL_VER ||= Kernel.system('which perl', %i[out err] => File::NULL) ? "perl#{`perl -MConfig -e "print \\\"\\\$Config{'PERL_API_REVISION'}.\\\$Config{'PERL_API_VERSION'}\\\";"`}" : 'perl5.40' unless defined?(CREW_PERL_VER)
 CREW_PY_VER ||= Kernel.system("#{CREW_PREFIX}/bin/python3 --version", %i[out err] => File::NULL) ? "py#{`python3 -c "print('.'.join(__import__('platform').python_version_tuple()[:2]))"`.chomp}" : 'py3.13' unless defined?(CREW_PY_VER)
 CREW_RUBY_VER ||= "ruby#{RUBY_VERSION.slice(/(?:.*(?=\.))/)}" unless defined?(CREW_RUBY_VER)
+CREW_VALID_BUILDSYSTEMS ||= %w[Autotools CMake Meson PERL Package Pip Python Qmake RUBY RUST]
```

Comment:
Could this loop through all the files in buildsystems and build the array dynamically?

Reply:
Better not to slow down every single crew operation to make that happen?

Reply:
Also, notice the capitalization... You can't just look at the filenames.

Reply:
```
From the buildsystems directory:
$ time for p in $(ls); do grep ^class $p|cut -d' ' -f2; done|xargs
Autotools CMake Meson PERL Pip Python Qmake RUBY

real	0m0.009s
user	0m0.003s
sys	0m0.010s
```

Reply:
Rubify?

Reply:
This doesn't work:
```ruby
CREW_VALID_BUILDSYSTEMS ||= Dir.glob("#{CREW_LIB_PATH}/lib/buildsystems/*.rb") { |file| File.readlines(file, chomp: true).each { |line| p line.split[1] if line =~ /^class/ } }
```
Any ideas?

@supechicken ?

Reply:
(I just get nil)

Reply:
Also doesn't work:
```ruby
CREW_VALID_BUILDSYSTEMS = Dir.glob("#{CREW_LIB_PATH}/lib/buildsystems/*.rb") { |file| Fil
e.readlines(file, chomp: true).each { |line| print "#{line.split[1]} " if line =~ /^class/ } }
```

Reply:
Ok this works:
```ruby
@buildsystems = []
Dir.glob("#{CREW_LIB_PATH}/lib/buildsystems/*.rb") { |file| @buildsystems << File.foreach(file).grep(/^class/).to_s.split[1] }
CREW_VALID_BUILDSYSTEMS ||= @buildsystems
```